core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSignerTest.java

```diff
@@ -343,4 +344,337 @@ public void sign_WithPayloadSigningNullAndHttp_FallsBackToPayloadSigning() {
         assertThat(signedRequest.request().firstMatchingHeader("x-amz-content-sha256"))
             .hasValue("a15c8292b1d12abbbbe4148605f7872fbdf645618fee5ab0e8072a7b34f155e2");
     }
+
+    @Test
```

Comment:
for code-organization, I wonder if keeping the sign/signAsync tests for each use case next to each other is better. Easier to compare both tests together, and guide new test writer to add tests for both cases.

Reply:
Maybe? But there may be slight differences in behavior between sync and async, so you'd end up with a weird mishmash of sync_xyz, async_abc_xyz, async_def_xyz, and so on.

Reply:
Imo, probably still better, as in majority of the cases, they'll be similar. And if there are multiple sub-cases in async compared to sync, seeing them together is also useful.

Either ways, it's a nit, and non-blocking.

Reply:
Sure, will re-order